pyroengine/engine/pyronear_engine.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (C) 2021, Pyronear contributors.
+
+# This program is licensed under the GNU Affero General Public License version 3.
+# See LICENSE or go to <https://www.gnu.org/licenses/agpl-3.0.txt> for full license details.
+
+from .pyronear_predictor import PyronearPredictor
+from pyroclient import client
+
+
+api_url = "http://pyronear-api.herokuapp.com"
+
+
+class PyronearEngine:
+    """
+    This class is the Pyronear Engine. This engine manage the whole Fire Detection
+    process by capturing and saving the image and by predicting if there is a fire or
+    not based on this image.
+    Example
+    -------
+    pyroEngine = PyronearEngine(api_login, api_password)
+    pyroEngine.run()
+    """
+    def __init__(self, api_login, api_password):
+        # Pyronear Predictor
+        self.pyronearPredictor = PyronearPredictor()
+
+        # API Setup
+        self.create_device(api_login, api_password)
+
+    def run(self):
+        """Should be implemented for each patform, with an adapted image capture method"""
+        pass
+
+    def predict(self, frame):
+
+        res = self.pyronearPredictor.predict(frame)
+        if res > 0.5:
+            print('Fire detected !!!')
+            frame.save('fire.jpg')
```

Comment:
We lose precious time here. Currently:
- if a wildfire is detected, the loaded image will be saved to disk
- then the api will load it from disk and upload it with the API client

A better option:
- if a wildfire is detected, pass the loaded image directly to send the alert (rather than writing + loading from disk)
- if needed, also save the image to disk